http_client.go

```diff
@@ -29,34 +31,41 @@ func userFriendlyErr(errStr string) string {
 	return "HttpErr"
 }
 
-func (h *HttpClient) PostJson(uri string, jsonBytes []byte) (statusCode int) {
+func (h *HttpClient) createPayloadRequest(uri string, jsonBytes []byte) (*http.Request, error) {
```

Comment:
added this function to simplify the PostJson routine.  Otherwise gocyclo complained about PostJson being "too complex", meh.